app/src/main/java/com/oracle/javafx/scenebuilder/app/SceneBuilderApp.java

```diff
@@ -112,12 +113,14 @@
     }
 
     private static SceneBuilderApp singleton;
-    private static final CountDownLatch launchLatch = new CountDownLatch(1);
 
     private final ObservableList<DocumentWindowController> windowList = FXCollections.observableArrayList();
     private UserLibrary userLibrary;
     private ToolTheme toolTheme = ToolTheme.DEFAULT;
 
+    private final ObservableList<Runnable> startupTasks = FXCollections.observableArrayList();
+    private final BooleanBinding startupTasksFinished = Bindings.isEmpty(startupTasks);
```

Comment:
@AlmasB this is not addressed yet

Reply:
Ypu can do:

```
private BooleanBinding startupTasksFinished;
...
public void handleLaunch(List<String> files) {
...
            startupTasksFinished = Bindings.isEmpty(startupTasks);
            startupTasksFinished.addListener((o, old, isFinished)  -> {
...
```
Or directly:
```
public void handleLaunch(List<String> files) {
...
            final BooleanBinding startupTasksFinished = Bindings.isEmpty(startupTasks);
            startupTasksFinished.addListener((o, old, isFinished)  -> {
...
```

Reply:
Happy to do (1), but not sure about the benefits over the current approach?

As for (2), do we have a guarantee that `startupTasksFinished` alongside its listener won't be GC'ed when the binding goes out of scope?

Reply:
1) avoids creating the binding object at the beginning. Note that it might be cases where it is not used at all (when `showWelcomeDialog` is false).
2) I have the same thoughts about the risk of the binging being GC'ed (I had bad experiences in the past with Bindings...), that's why I prefer 1).

Reply:
Ok, that clears it up for (1), will do.